hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/DistributedObjectCounterCollector.java

```diff
@@ -24,59 +24,31 @@
 import com.hazelcast.multimap.impl.MultiMapService;
 import com.hazelcast.ringbuffer.impl.RingbufferService;
 
+
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.stream.Collectors;
 
-import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.groupingBy;
 
-class DistributedObjectCounterCollector implements MetricsCollector {
 
-    Collection<DistributedObject> maps;
-    Collection<DistributedObject> sets;
-    Collection<DistributedObject> queues;
-    Collection<DistributedObject> multimaps;
-    Collection<DistributedObject> lists;
-    Collection<DistributedObject> ringbuffers;
+class DistributedObjectCounterCollector implements MetricsCollector {
 
 
     @Override
     public Map<String, String> computeMetrics(Node hazelcastNode) {
-        final int[] count = {0, 0, 0, 0, 0, 0};
 
         Collection<DistributedObject> distributedObjects = hazelcastNode.hazelcastInstance.getDistributedObjects();
-
-        distributedObjects.stream().forEach(distributedObject -> {
-            switch (distributedObject.getServiceName()) {
-                case MapService.SERVICE_NAME:
-                    count[0]++;
-                    break;
-                case SetService.SERVICE_NAME:
-                    count[1]++;
-                    break;
-                case QueueService.SERVICE_NAME:
-                    count[2]++;
-                    break;
-                case MultiMapService.SERVICE_NAME:
-                    count[3]++;
-                    break;
-                case ListService.SERVICE_NAME:
-                    count[4]++;
-                    break;
-                case RingbufferService.SERVICE_NAME:
-                    count[5]++;
-                    break;
-            }
-        });
-
+        Map<String, Long> result = distributedObjects.stream().collect(groupingBy(DistributedObject::getServiceName, Collectors.counting()));
```

Comment:
This looks much nicer than before, thank you. One concern at the moment is that it will count all datastructures, including the ones that we don't care for now, so please filter those out in an intermediate `filter()` operation. Thanks.